src/macros.rs

```diff
@@ -239,103 +255,117 @@ fn rewrite_macro_inner(
+                        MacroPosition::Item => format!("{};", rw),
+                        _ => rw,
+                    })
+                }
+            }
+            Delimiter::Bracket => {
+                // Handle special case: `vec![expr; expr]`
+                if vec_with_semi {
+                    handle_vec_semi(context, shape, arg_vec, macro_name, style)
+                } else {
+                    // If we are rewriting `vec!` macro or other special macros,
+                    // then we can rewrite this as a usual array literal.
+                    // Otherwise, we must preserve the original existence of trailing comma.
+                    let mut force_trailing_comma = if trailing_comma {
                         Some(SeparatorTactic::Always)
                     } else {
                         Some(SeparatorTactic::Never)
-                    },
-                )
-                .map(|rw| match position {
-                    MacroPosition::Item => format!("{};", rw),
-                    _ => rw,
-                })
-            }
-        }
-        Delimiter::Bracket => {
-            // Handle special case: `vec![expr; expr]`
-            if vec_with_semi {
-                handle_vec_semi(context, shape, arg_vec, macro_name, style)
-            } else {
-                // If we are rewriting `vec!` macro or other special macros,
-                // then we can rewrite this as a usual array literal.
-                // Otherwise, we must preserve the original existence of trailing comma.
-                let mut force_trailing_comma = if trailing_comma {
-                    Some(SeparatorTactic::Always)
-                } else {
-                    Some(SeparatorTactic::Never)
-                };
-                if is_forced_bracket && !is_nested_macro {
-                    context.leave_macro();
-                    if context.use_block_indent() {
-                        force_trailing_comma = Some(SeparatorTactic::Vertical);
                     };
+                    if is_forced_bracket && !is_nested_macro {
+                        context.leave_macro();
+                        if context.use_block_indent() {
+                            force_trailing_comma = Some(SeparatorTactic::Vertical);
+                        };
+                    }
+                    rewrite_array(
+                        &macro_name,
+                        arg_vec.iter(),
+                        mac.span(),
+                        context,
+                        shape,
+                        force_trailing_comma,
+                        Some(original_style),
+                    )
+                    .map(|rewrite| {
+                        let comma = match position {
+                            MacroPosition::Item => ";",
+                            _ => "",
+                        };
+
+                        format!("{rewrite}{comma}")
+                    })
```

Comment:
Let's leverage `let-else` to help us with this one. it makes it easier to return early when rewriting fails.

```rust
let Some(rewrite) = rewrite_array(
    /* argument go here */
) else {
    return (None, None)
};
```